Where_Is_My_Wife/Assets/_Main/Scripts/SaveSystem/DataSaveManager.cs

```diff
@@ -48,32 +50,40 @@ private void Save()
 
         private void Load()
         {
-            string json = PlayerPrefs.GetString(SaveKey,null);
+            string json = PlayerPrefs.GetString(SaveKey, null);
             if (json == null || json == "")
             {
                 savedData = new Dictionary<string, object>();
                 Debug.Log("No save data found. Created new one.");
             }
             else
             {
-                savedData = JsonConvert.DeserializeObject<Dictionary<string, object>>(json); 
+                savedData = JsonConvert.DeserializeObject<Dictionary<string, object>>(json);
                 Debug.Log("Game data loaded properly.");
             }
         }
 
         public void DeleteSaveData()
         {
-            SetData(LastUnlockedLevelNumberKey, 0);
-            SetData(LastUnlockedLevelSceneNameKey, null);
+            SetData(LastPlayedStoryModeLevelSceneNameKey, null);
         }
 
-        public void SetLastUnlockedLevel(int levelNumber, string sceneName)
+        public void SetNextLevelParameters(int levelNumber, string sceneName)
         {
             if (levelNumber > GetData<int>(LastUnlockedLevelNumberKey))
             {
                 SetData(LastUnlockedLevelNumberKey, levelNumber);
-                SetData(LastUnlockedLevelSceneNameKey, sceneName);
             }
+
+            if (IsInStoryMode)
+            {
+                SetData(LastPlayedStoryModeLevelSceneNameKey, sceneName);
+            }
+        }
+        
+        public void SetStoryMode(bool isInStoryMode)
+        {
+            IsInStoryMode = isInStoryMode;
```

Comment:
Move this to `LevelManager` and add a parameter in `SetNextLevelParameters` to specify if is in story mode or not.